cmd/attack/kafka.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2022 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package attack
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"go.uber.org/fx"
+
+	"github.com/chaos-mesh/chaosd/cmd/server"
+	"github.com/chaos-mesh/chaosd/pkg/core"
+	"github.com/chaos-mesh/chaosd/pkg/server/chaosd"
+	"github.com/chaos-mesh/chaosd/pkg/utils"
+)
+
+func NewKafkaAttackCommand(uid *string) *cobra.Command {
+	options := core.NewKafkaCommand()
+	dep := fx.Options(
+		server.Module,
+		fx.Provide(func() *core.KafkaCommand {
+			options.UID = *uid
+			return options
+		}),
+	)
+
+	cmd := &cobra.Command{
+		Use:   "kafka <subcommand>",
+		Short: "Kafka attack related commands",
+	}
+
+	cmd.PersistentFlags().StringVarP(&options.Topic, "topic", "T", "", "the topic to attack")
+	cmd.PersistentFlags().IntVarP(&options.Partition, "partition", "", 0, "the partition to attack")
```

Comment:
if the partition is 0(default), chaosd will only write messages to this partition, is it right?

Reply:
Yes

Reply:
Writing messages to only one partition will reduce the performance. How about using `-1` as the default value, and when the partition is -1,  write messages to the topic and without specify a partition

Reply:
Oops, I do not find relevant logic in the docs or source code of [` LookupPartition`](https://github.com/segmentio/kafka-go/blob/v0.4.31/dialer.go#L162) and [`DialLeader`](https://github.com/segmentio/kafka-go/blob/v0.4.31/dialer.go#L132)

Reply:
Done